packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,389 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/process.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:meta/meta.dart';
+import 'package:vm_snapshot_analysis/treemap.dart';
+
+import '../base/file_system.dart';
+import '../convert.dart';
+import '../globals.dart' as globals;
+
+class SizeAnalyzer {
+  static const String aotSizeFileName = 'aot-size.json';
+
+  static String aotSizeAnalysisJsonFile(String aotOutputPath) {
+    return globals.fs.path.join(aotOutputPath, aotSizeFileName);
+  }
+
+  static String aotSizeAnalysisExtraGenSnapshotOption(String aotOutputPath) {
+    return '--print-instructions-sizes-to=${aotSizeAnalysisJsonFile(aotOutputPath)}';
+  }
+
+  static const int tableWidth = 80;
+
+  Map<String, dynamic> analyzeApkSize({
+    @required File apk,
+    @required File aotSizeJson,
+  }) {
+    globals.printStatus('▒' * tableWidth);
+    _printEntitySize(
+      '${apk.basename} (total compressed)',
+      apk.lengthSync(),
+      0,
+      showColor: false,
+    );
+    globals.printStatus('━' * tableWidth);
+    final Directory tempApkContent = globals.fs.systemTempDirectory.createTempSync('flutter_tools.');
+    // TODO: implement Windows.
+    String unzipOut;
+    try {
+      unzipOut = processUtils.runSync(<String>[
+        'unzip',
```

Comment:
this can just be zipinfo -l rather than actually extracting the files into a temp directory